core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -189,6 +192,36 @@ void testRemoteLogMetadataManagerWithUserDefinedConfigs() {
         assertFalse(metadataMangerConfig.containsKey("remote.log.metadata.y"));
     }
 
+    @Test
```

Comment:
perhaps we can add a test to verify that an exception is thrown when endpoint is not defined?

Reply:
Good idea. Added!